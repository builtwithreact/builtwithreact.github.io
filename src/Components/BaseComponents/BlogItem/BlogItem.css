.BlogItem {
  display: flex; }
  .BlogItem .BlogItem__Image {
    width: 250px;
    height: 100%;
    margin-right: 40px; }
  .BlogItem .BlogItem__Body {
    display: flex;
    flex-direction: column;
    flex: 1; }
  .BlogItem .BlogItem__Body__Content__Link {
    border-top: 1px #ccc solid;
    margin-top: auto;
    text-transform: uppercase; }
