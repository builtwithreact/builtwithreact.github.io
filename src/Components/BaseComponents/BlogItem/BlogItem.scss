.BlogItem {
  display: flex;
  .BlogItem__Image {
    width: 250px;
    height: 100%;
    margin-right: 40px;
  }
  .BlogItem__Body {
    display: flex;
    flex-direction: column;
    flex: 1;
  }

  .BlogItem__Body__Content { 
  }

  .BlogItem__Body__Content__Link {
    border-top: 1px #ccc solid;
    margin-top: auto;
    text-transform: uppercase;
  }
}