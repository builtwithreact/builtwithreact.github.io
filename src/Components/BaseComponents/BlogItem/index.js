import React from 'react';
import PropTypes from 'prop-types';
import sample from './sample.jpg'
import './BlogItem.css';

const BlogItem = (props) => {
    return (
        <div className="BlogItem">
            <img src={props.imageUrl} alt={props.title} className="BlogItem__Image" />
            <div className="BlogItem__Body">
                <h2 className="BlogItem__Body__Title">
                    {props.title}
                </h2>
                <div className="BlogItem__Body__Content">
                    {props.content}
                    {/*  */}
                </div>
                <div className="BlogItem__Body__Content__Link">
                    <a href={props.readMoreLink}> Read more </a>
                </div>
            </div>
        </div>
    )
}

BlogItem.propTypes = {
    title: PropTypes.string.isRequired,
    imageUrl : PropTypes.string,
    content : PropTypes.string,
    readMoreLink : PropTypes.string
}

BlogItem.defaultProps = {
    imageUrl: sample
}

export default BlogItem;