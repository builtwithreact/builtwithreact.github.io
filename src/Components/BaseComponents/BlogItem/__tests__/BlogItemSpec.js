import React from 'react';
import renderer from 'react-test-renderer';
import BlogItem from './../index';

describe('<BlogItem />', () => {
    it('should render correctly', () => {
        const blogItemProps = {
            title: "some title",
            content: "some content",
            imageUrl: "https://someimage.url",
            readMoreLink: "https://someremorelink.url",
        }
        let tree = renderer.create(
            <BlogItem  {...blogItemProps}/>
        ).toJSON();

        expect(tree).toMatchSnapshot()
    })

    it('should render with default image if not provided', () => {
        const blogItemProps = {
            title: "some title",
            content: "some content",
            readMoreLink: "https://someremorelink.url",
        }
        let tree = renderer.create(
            <BlogItem  {...blogItemProps}/>
        ).toJSON();

        expect(tree).toMatchSnapshot()
    })

})