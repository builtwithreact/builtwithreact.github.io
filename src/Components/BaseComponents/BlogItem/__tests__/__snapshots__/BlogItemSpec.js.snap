// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogItem /> should render correctly 1`] = `
<div
  className="BlogItem"
>
  <img
    alt="some title"
    className="BlogItem__Image"
    src="https://someimage.url"
  />
  <div
    className="BlogItem__Body"
  >
    <h2
      className="BlogItem__Body__Title"
    >
      some title
    </h2>
    <div
      className="BlogItem__Body__Content"
    >
      some content
    </div>
    <div
      className="BlogItem__Body__Content__Link"
    >
      <a
        href="https://someremorelink.url"
      >
        Read more
      </a>
    </div>
  </div>
</div>
`;

exports[`<BlogItem /> should render with default image if not provided 1`] = `
<div
  className="BlogItem"
>
  <img
    alt="some title"
    className="BlogItem__Image"
    src="sample.jpg"
  />
  <div
    className="BlogItem__Body"
  >
    <h2
      className="BlogItem__Body__Title"
    >
      some title
    </h2>
    <div
      className="BlogItem__Body__Content"
    >
      some content
    </div>
    <div
      className="BlogItem__Body__Content__Link"
    >
      <a
        href="https://someremorelink.url"
      >
        Read more
      </a>
    </div>
  </div>
</div>
`;
